java/src/main/java/com/google/oak/client/OakClient.java

```diff
@@ -42,15 +51,24 @@ public class OakClient<T extends Transport> implements AutoCloseable {
    * @return an instance of the {@code OakClient} wrapped in a {@code Result}
    */
   public static <E extends EvidenceProvider & Transport, V extends AttestationVerifier>
-      Result<OakClient<E>, Exception> create(E transport, V verifier) {
+      Result<OakClient<E>, Exception> create(
+          E transport, V verifier, Clock clock, ReferenceValues referenceValues) {
     // TODO(#3641): Implement client-side attestation verification.
     return transport.getEvidence()
         .mapError(Exception::new)
         .andThen(e
-            -> verifier.verify(e.getAttestationEvidence(), e.getAttestationEndorsement())
+            -> verifier
+                   .verify(clock.instant().toEpochMilli(), e.getEvidence(), e.getEndorsements(),
+                       referenceValues)
                    .map(b
                        -> new OakClient<E>(transport,
-                           e.getAttestationEvidence().getEncryptionPublicKey().toByteArray())));
+                           e.hasEvidence() ? e.getEvidence()
+                                                 .getApplicationKeys()
+                                                 .getEncryptionPublicKeyCertificate()
+                                                 .toByteArray()
+                                           : e.getAttestationEvidence()
```

Comment:
Does it mean that we currently will also handle the old non-DICE evidence?

Reply:
Yeah, for now - until we also update the transport layer. Let's sync on how to best handle this, I'm not sure how much of server code we want to support in Java